moviebot/controller/controller_flask.py

```diff
@@ -0,0 +1,398 @@
+
+        Returns:
+            Object with start message to send to the server.
+        """
+        self.agent[user_id] = Agent(self.configuration)
+
+        (
+            self.agent_response[user_id],
+            self.record_data_agent[user_id],
+            self.user_options[user_id],
+        ) = self.agent[user_id].start_dialogue()
+        if restart:
+            (
+                self.agent_response[user_id],
+                self.user_options[user_id],
+                self.record_data_agent[user_id],
+            ) = self.agent[user_id].start_dialogue(None, restart)
+            return self.user_messages[user_id].text(
+                self.agent_response[user_id], intent=self.agent_intent
+            )
+        else:
+            return {"message": {"text": "", "intent": self.agent_intent}}
+
+    def get_movie_id(self, response: str) -> str:
+        """Retrieves movie id from agent response string.
+
+        Args:
+            response: Agent response.
+
+        Returns:
+            Movie id.
+        """
+        if "/tt" in response:
+            start = response.find("/tt")
+            movie_id = response[start + 3 : start + 10]
+            return movie_id
+
+    def continue_dialogue(self, user_id: str, payload: str) -> None:
+        """Continues dialogue with the agent. Updates movie info.
+
+        Args:
+            user_id: User id.
+            payload: User payload.
+        """
+        user_utterance = UserUtterance({"text": payload})
+        (
+            self.agent_response[user_id],
+            self.user_options[user_id],
+            self.record_data_agent[user_id],
+        ) = self.agent[user_id].continue_dialogue(
+            user_utterance, self.user_options[user_id]
+        )
+        if self.users[user_id]:
+            self.record(user_id, payload)
+        movie_id = self.get_movie_id(self.agent_response[user_id])
+        self.movie_info(movie_id, user_id)
+
+    def store_user(self, user_id: str) -> Callable:
+        """True if user accepts license agreement. Data for this user is stored
+        in conversation history.
+
+        Args:
+            user_id: User id.
+        """
+        self.users[user_id] = True
+        return self.store_data(user_id)
+
+    def record(self, user_id: str, payload: str) -> None:
+        """Records user conversation if user has accepted privacy policy.
+
+        Args:
+            user_id: User id.
+            payload: User payload.
+        """
+        if self.agent[user_id].bot_recorder:
+            self.record_data[user_id] = {
+                "Timestamp": str(datetime.now()),
+                "User_Input": payload,
+            }
+            self.record_data[user_id].update(self.record_data_agent[user_id])
+            self.agent[user_id].bot_recorder.record_user_data(
+                user_id, self.record_data[user_id]
+            )
+
+    def load_user_data(self, user_id: str) -> None:
```

Comment:
Consider moving this to the `Controller` class.

Reply:
Done, along with the method to delete conversation history.